app-editors/scite/scite-4.1.2.ebuild

```diff
@@ -0,0 +1,156 @@
+	x11-libs/pango
+	lua? ( >=dev-lang/lua-5:= )
+	clang? ( >=sys-devel/clang-5:= )
+"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S="${WORKDIR}/${PN}/gtk"
+
+pkg_pretend() {
+	# use clang could be package specific
+	if use clang && !(tc-is-clang) ; then
+		local clangMajorVersion="$( clang++ -dM -E -x c /dev/null |
+			 grep __clang_major | grep -o -E '[[:digit:]]' )"
+
+		# need C++17 features
+		# FIXME: Better Check of C++17 Features
+		[[ "${MERGE_TYPE}" != "binary" &&
+		${clangMajorVersion} -lt 5 ]] &&
+		die "Sorry, SCiTE uses C++17 Features and needs >sys-devel/clang-5
+		($(clangMajorVersion))."
+
+	# global choice of clang
+	elif tc-is-clang ; then
+		# need c++17 features
+		[[ "${MERGE_TYPE}" != "binary" &&
+		$(clang-major-version) -lt 5 ]] &&
+		die "Sorry, SCiTE uses C++17 Features and needs >sys-devel/clang-5
+		($(clang-major-version))."
+
+	# standard: gcc
+	elif tc-is-gcc; then
+		# older gcc is not supported
+		[[ "${MERGE_TYPE}" != "binary" &&
+		$(gcc-major-version) -lt 7 ]] &&
+		die "Sorry, Scite uses C++17 Features, need >sys-devel/gcc-7."
+	else
+		die "Either gcc or clang should be available for building scite"
+	fi;
+}
+
+pkg_setup() {
+	# presumably needed and maybe useful
+	[ use clang || tc-is-clang ] && llvm_pkg_setup;
+}
+
+src_prepare() {
+	# inheritet from earlier versions, maybe obsolete
+	sed -i "${WORKDIR}/scintilla/gtk/makefile" \
+		-e "s#-Os##" \
+		|| die "error patching /scintilla/gtk/makefile"
+
+	sed -i "${S}/makefile" \
+		-e "s#-rdynamic#-rdynamic ${LDFLAGS}#" \
+		-e 's#${D}##' \
+		-e "s#-Os##" \
+		|| die "error patching gtk/makefile"
+
+	# clang-6 complained about unknown warnings
+	if use clang || tc-is-clang ; then
+		# remove unknown warning options
+		sed -i "${WORKDIR}/scintilla/gtk/makefile" \
+			-e "s#-Wno-misleading-indentation##" \
+		|| die "error patching /scintilla/gtk/makefile for clang"
+
+		sed -i "${S}/makefile" \
+			-e "s#-Wno-misleading-indentation##" \
+		|| die "error patching gtk/makefile for clang"
+	fi;
+
+	# repair and enhance the .desktop file
+	sed -i "${S}/SciTE.desktop" \
+		-e "s#text/plain#text/\*;application/xhtml+xml#" \
+		-e "s#^Categories=\(.*\)#Categories=Development;#" \
+		|| die "error patching /scite/gtk/SciTe.desktop"
+
+	#  add the ebuild suffix as shell type for working with ebuilds
+	sed -i "${WORKDIR}/scite/src/perl.properties" \
+		-e "s#\*.sh;\*.bsh;#\*.ebuild;\*.sh;\*.bsh;#" \
+		|| die "error patching /scite/src/perl.prperties"
+
+	# it seems that pwd here is ${S}, but user patches are relative to ${workdir}
+	# Bug #576162
+
+	cd "${WORKDIR}" || die "couldn't cd into ${WORKDIR}"
+	eapply_user
+}
+
+src_configure() {
+	# seen in thunderbird ebuild, seems reasonable
+	if use clang && ! tc-is-clang ; then
+		# force clang
+		einfo "using clang..."
+		CC=${CHOST}-clang
+		CXX=${CHOST}-clang++
+		strip-unsupported-flags
+	fi
+}
+
+src_compile() {
+	# prepare make options
+	local emake_pars="GTK3=1"
+
+	if [ use clang || tc-is-clang ]; then
```

Comment:
I feel like only checking for clang USE is enough here too, since you force it on earlier. You can delete the brackets and check for tc.